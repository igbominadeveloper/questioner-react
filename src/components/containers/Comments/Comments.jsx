import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';

//components
import Comment from '../../presentationals/Comment/Comment';
import Loader from '../../presentationals/Loader/Loader';

//modules
import { checkAndRedirect } from '../../../store/modules/auth';
import { createNewComment } from '../../../store/modules/question';
export class Comments extends Component {
  state = {
    comment: '',
  };

  inputHandler = event => {
    this.setState({
      [event.target.name]: event.target.value,
    });
  };

  commentSubmitHandler = event => {
    const { isAuthenticated, location, history } = this.props;
    event.preventDefault();
    if (!isAuthenticated) {
      return this.props.checkAndRedirect(location.pathname, history);
    }
    this.props.createNewComment({
      question_id: this.props.question,
      comment: this.state.comment,
    });
    this.setState({ comment: '' });
  };

  render() {
    const { comments, isLoading } = this.props;
    return (
      <div className="question-comments">
        {isLoading && <Loader />}
        {comments &&
          comments.length > 0 &&
          comments.map(comment => (
            <Comment comment={comment} key={comment.id} />
          ))}
        <div className="new-comment">
          <form
            action=""
            id="comment-form"
            className="flex flex-column"
            onSubmit={this.commentSubmitHandler}
          >
            <div className="form-group">
              <textarea
                className="form-control"
                name="comment"
                id="comment"
                value={this.state.comment}
                placeholder="Your comment"
                required
                onChange={this.inputHandler}
              />
            </div>
            <button className="btn-primary p-1 shadow">post</button>
          </form>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isAuthenticated: state.auth.token !== null,
  isLoading: state.question.isLoading,
});

export default connect(
  mapStateToProps,
  { checkAndRedirect, createNewComment },
)(withRouter(Comments));
