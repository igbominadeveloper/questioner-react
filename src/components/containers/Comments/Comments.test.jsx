import React from 'react';
import { shallow } from 'enzyme';

import { Comments } from './Comments';
const props = {
  comment: [],
};

const component = shallow(<Comments {...props} />);

describe('Comments', () => {
  it('renders without errors', () => {
    expect(component).toMatchSnapshot();
  });
});
