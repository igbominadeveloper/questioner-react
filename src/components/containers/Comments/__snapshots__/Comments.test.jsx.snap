// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments renders without errors 1`] = `
<div
  className="question-comments"
>
  <div
    className="new-comment"
  >
    <form
      action=""
      className="flex flex-column"
      id="comment-form"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <textarea
          className="form-control"
          id="comment"
          name="comment"
          onChange={[Function]}
          placeholder="Your comment"
          required={true}
          value=""
        />
      </div>
      <button
        className="btn-primary p-1 shadow"
      >
        post
      </button>
    </form>
  </div>
</div>
`;
